### PR TITLE
IATI rulesets fixes

### PR DESCRIPTION
**Pull request by Bjwebb**
* Test multiple elements when appropriate #868 
* Check for @ref or narrative/text() in every participating-org #870 
* Add recipient-country|recipient-region percentage sum check #871

### CODE REVIEW

File: cove_iati/rulesets/iati_standard_v2_ruleset/steps/standard_ruleset_step_definitions.py [hunk 1]
```diff
@@ -26,28 +27,28 @@ def step_given_elements(context, xpath_expression):
...
-
-
-@then('`{attribute}` attribute must be today or in the past')
+@then('`{xpath}` must be today or in the past')
 @register_ruleset_errors()
-def step_must_be_today_or_past(context, attribute):
+def step_must_be_today_or_past(context, xpath):
     errors = []
     today = datetime.date.today()
+    parents = get_xobjects(context.xml, context.xpath_expression)
```
Comment (edugomez) [inline cove_iati/rulesets/iati_standard_v2_ruleset/steps/standard_ruleset_step_definitions.py]: So if the ```context.xpath_expression``` is ```'.'``` (which you have changed in few places) then we just get ```context.xml``` back within a array, right?

Comment (Bjwebb) [reply cove_iati/rulesets/iati_standard_v2_ruleset/steps/standard_ruleset_step_definitions.py]: Yes

File: cove_iati/rulesets/iati_standard_v2_ruleset/actual_dates_in_the_past.feature [hunk 2]
```diff
@@ -1,17 +1,17 @@
 Feature: Actual dates must be in the past
 
   Scenario Outline: activity-date[@type='2'].@iso-date must be today or in the past
-    Given `activity-date[@type='2']` elements
-      Then `iso-date` attribute must be today or in the past
+    Given `.` elements
+      Then `activity-date[@type='2']/@iso-date` must be today or in the past
```
Comment (edugomez) [inline cove_iati/rulesets/iati_standard_v2_ruleset/actual_dates_in_the_past.feature]: This is nice if it does what I think it does in the rule step function (see my comment https://github.com/OpenDataServices/cove/pull/869/files#r145083188) as it keeps concise and quite readable.

File: cove_iati/rulesets/iati_standard_v2_ruleset/steps/standard_ruleset_step_definitions.py [hunk 3]
```diff
@@ -142,30 +148,48 @@ def step_not_expected(context, xpath_expression):
...
+                path_attr1 = get_child_full_xpath(context.xml, start_date_attr)
+                path_attr2 = get_child_full_xpath(context.xml, end_date_attr)
+                errors.append({
+                    'message': fail_msg.format(start_date_attr, end_date_attr),
+                    'path': '{} & {}'.format(path_attr1, path_attr2)
+                })
+    return context, errors
+
+
+@then(u'`{attribute}` attribute must sum to 100')
```
Comment (edugomez) [inline cove_iati/rulesets/iati_standard_v2_ruleset/steps/standard_ruleset_step_definitions.py]: Is the scenario outline for this step missing (forgot to add a file)?

Comment (Bjwebb) [reply cove_iati/rulesets/iati_standard_v2_ruleset/steps/standard_ruleset_step_definitions.py]: d'oh

Comment (Bjwebb) [reply cove_iati/rulesets/iati_standard_v2_ruleset/steps/standard_ruleset_step_definitions.py]: I've rebased to include it.

Comment (edugomez) [reply cove_iati/rulesets/iati_standard_v2_ruleset/steps/standard_ruleset_step_definitions.py]: Cool. I just pulled the rebased branch and spotted a tiny typo in a feature file (older, NOT added by your code), I'm fixing and pushing it.

**Pull request merged**